modules/pel/peltool/peltool.py

```diff
@@ -267,7 +267,82 @@ def parseAndWriteOutput(file: str, output_dir: str, config: Config,
             print(f"No PEL parsed for {file}: {e}")
 
 
+def parsePELSummary(stream: DataStream, config: Config, critSysTerm:bool = False):
+    """
+    Parses and summarizes data from a PEL stream.
+    Returns: Event ID (eid) and summary of PEL data.
+            If no PEL is parsed or an error occurs, empty strings are returned.
+    """
+    out = OrderedDict()
+    ret, ph = generatePH(stream, out)
+    if ret is False:
+        return "", ""
+    eid = ph.pLID
+    ret, uh = generateUH(stream, ph.creatorID, out)
+    if ret is False:
+        return "", ""
+    if not uh.isServiceable():
+        return "", ""
+    
+    summary = OrderedDict()
+    for _ in range(2, ph.sectionCount):
+        sectionID, sectionLen, versionID, subType, componentID = parseHeader(stream)
+        section_json = OrderedDict()
+        sectionFun(stream, section_json, sectionID, sectionLen,
+                   versionID, subType, componentID, ph.creatorID, config)
+        if sectionID == SectionID.primarySRC.value:
+            summary["SRC"] = section_json["Primary SRC"]["Reference Code"]
+            if "Error Details" in section_json["Primary SRC"] :
+                summary["Message"] = section_json["Primary SRC"]["Error Details"]["Message"]
+            break
+    summary["PLID"] = ph.pLID
+    summary["CreatorID"] = out["Private Header"]["Creator Subsystem"]
+    summary["Subsystem"] = out["User Header"]["Subsystem"]
+    summary["Commit Time"] = ph.commitTime
+    summary["Sev"] = out["User Header"]["Event Severity"]
+    summary["CompID"] = out["Private Header"]["Created by"]
+    return eid, summary
+
+def extractAndSummarizePEL(file: str, config: Config,
+                        delete_after_parsing: bool):
+    """
+    Extracts and summarizes data from a PEL file.
+    Returns: Event ID (eid) and summary extracted from the PEL.
+            If no PEL is parsed, empty strings are returned.
+    """
+    with open(file, 'rb') as fd:
+        data = fd.read()
+        stream = DataStream(data, byte_order='big', is_signed=False)
+        try:
+            eid, summary = parsePELSummary(stream, config)
+            if eid :
+                return eid, summary
+        except Exception as e:
+            print(f"Exception: No PEL parsed for {file}: {e}")
```

Comment:
Do you have any test result for the negative case, I would like to make sure how are we printing error details if some PEL is not able to parse and PEL summary of few PELs if able to parse.

Reply:
Yeah, For a file with text content:  `PH test test` , 
Got this before full output:

 `Exception: No PEL parsed for testfolder/errp: range check failure`

Reply:
@spinler are you ok if we printed errors PELs?

Reply:
Ideally no, though for example if that PH wasn't there some other code would print an error already.
Theoretically this shouldn't happen because the PEL daemon will already delete bad PELs, but who knows.

What if we added a --verbose option and then print the error here if set?  I think that wouldn't be too bad to add?  just set a new flag on Config?

Reply:
If we add --verbose, we will also want to put all other error message under this. 
Like [LN42](https://github.com/ibm-openbmc/openpower-pel-parsers/blob/master/modules/pel/peltool/peltool.py#L42), LN55, etc.
This could be a functionality in itself.

Reply:
We could handle it separately for the whole tool by adding verbose option.